app/models/payment_additional_info.rb

```diff
@@ -1,13 +1,48 @@
-require "flex_commerce_api/api_base"
 module FlexCommerce
   #
-  # A flex commerce payment additional info model
+  # A flex commerce PaymentAdditionalInfo model
   #
-  # The content depends on the payment gateway being used.  For paypal for example,
-  # we will get address details and shipping method back from shift
+  # This will be triggered once user returns back to paypal confirmation page.
+  # This model will get address details and shipping method back from paypal.
   #
-  #
-  class PaymentAdditionalInfo < FlexCommerceApi::ApiBase
+  class PaymentAdditionalInfo
+    include ActiveModel::Model
+    attr_accessor :meta, :id, :options
+
+    # @method call
+    # 
+    # Triggers call to Paypal Express and fetch the info about the 
+    # billing, shipping address and shipping method details
+    # 
+    # @return [FlexCommerce::PaymentAdditionalInfo]
+    def call
+      apply_gateway_response_filter.first
+    end
+
+    def records(*)
+      []
+    end
+
+    def meta_attributes
+      _model.meta.inject({}) do |acc, (key, value)|
+        acc.merge key => { value: value, data_type: value.class.name.underscore }
+      end
+    end
+
+    def meta_attribute(key)
+      begin
+        self.meta[key]
+      rescue NoMethodError => e
+        nil
+      end
+    end
+
+    private
+
+    def apply_gateway_response_filter
+      # Call AdditionalInfo service class to trigger Paypal for details
+      records << ::FlexCommerce::Payments::PaypalExpress::AdditionalInfo.new(options: options).call
```

Comment:
Drop this entire class, we can have the  `FlexCommerce::Payments::PaypalExpress` call `FlexCommerce::Payments::PaypalExpress::AdditionalInfo` directly.